docs/docs/overview/key_terms.md

```diff
@@ -28,14 +28,20 @@ permalink: docs/overview/key_terms.html
 
 ## Secretless Broker-Specific Terminology
 
-- ### Listener
+- ### <a href="/docs/reference/listeners.html">Listeners</a>
 
   A Listener listens on a TCP port or Unix socket for connections from clients. When a connection is received, it's handed off to a Handler for further processing.
 
-- ### Provider
+- ### <a href="/docs/reference/providers/overview.html">Credential Providers</a>
 
-  The Secretless Broker can obtain credentials from many different vaults. For each kind of vault, the Secretless Broker contains a Provider which can connect to that vault and obtain credentials. By adding new Providers, the Secretless Broker can be extended to support new vaults.
+  The Secretless Broker can obtain credentials from many different vaults. For each kind of vault, the Secretless Broker contains a Credential Provider which can connect to that vault and obtain credentials. By adding new Credential Providers, the Secretless Broker can be extended to support new vaults.
```

Comment:
suggestion without strong justification: "many different" -> "a variety of", unless there is need for emphasis on the "many"